### PR TITLE
Unify RT Info And Add Methods To Output class

### PR DESCRIPTION
**Pull request by GlebKazantaev**
### Description
* Unified rt_info between Output descriptor and Tensor
* Added helpful methods to Output class

### Ticket
XXX-67353

### CODE REVIEW

File: ngraph/core/include/openvino/core/node_output.hpp [hunk 1]
```diff
@@ -77,6 +77,13 @@ class OPENVINO_API Output<Node> {
     /// \return The constant reference to runtime info map
     const RTMap& get_rt_info() const;
 
+    /// \return The tensor names associated with this output
+    const std::unordered_set<std::string>& get_names() const;
+    /// \return Set tensor names associated with this output
+    void set_names(const std::unordered_set<std::string>& names);
+    /// \return Add tensor names associated with this output
+    void add_names(const std::unordered_set<std::string>& names);
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/node_output.hpp:85]: will we have:
```c++
std::string get_name() const { return *get_names().begin(); }
```
?

Comment (ilya-lavrenov) [reply ngraph/core/include/openvino/core/node_output.hpp:85]: But if I need any name, it would be nice to have a short cut for this, IMO.

Comment (ilya-lavrenov) [reply ngraph/core/include/openvino/core/node_output.hpp:85]: See https://github.com/openvinotoolkit/openvino/pull/7829#discussion_r721353720 and ticket XXX-67353

Comment (ilyachur) [reply ngraph/core/include/openvino/core/node_output.hpp:85]: We still use get_name method. We shouldn't change the behavior of this method.

**Pull request merged**